File: php/scripts/update_cost_center.php
<?php
require_once(__DIR__."/../vendor/autoload.php");
require_once("utils.php");
use Namelivia\TravelPerk\ServiceProvider;
$isSandbox = false;
$travelperk = (new ServiceProvider())->build(getenv("API_KEY"), false);
$originalCostCenter = $travelperk->costCenters()->costCenters()->get("1");
$costCenter = $travelperk->costCenters()->costCenters()->modify("1")->setName("Updated name by api probe")->save();
write_output("updated_cost_center", [
	$costCenter->id,
	$costCenter->name,
	$costCenter->archived,
	$costCenter->countUsers,
	$costCenter->users,
    $costCenter->users[0]->firstName,
    $costCenter->users[0]->lastName,
    $costCenter->users[0]->email,
    $costCenter->users[0]->id,
    $costCenter->users[0]->phone,
    $costCenter->users[0]->profilePicture,
]);
$travelperk->costCenters()->costCenters()->modify("1")->setName($originalCostCenter->name)->save();

File: python/scripts/update_cost_center.py
import os
from travelperk_http_python.builder.builder import build
from utils import write_output
travelperk = build(os.environ["API_KEY"], False)
original_cost_center = travelperk.cost_centers().cost_centers().get("1")
cost_center = travelperk.cost_centers().cost_centers().modify("1").set_name(
    "Updated name by api probe"
).save()
write_output("updated_cost_center", [
    cost_center.id,
    cost_center.name,
    cost_center.archived,
    cost_center.count_users,
    len(cost_center.users),
    cost_center.users[0].first_name,
    cost_center.users[0].last_name,
    cost_center.users[0].email,
    cost_center.users[0].id,
    cost_center.users[0].phone,
    cost_center.users[0].profile_picture,
])
travelperk.cost_centers().cost_centers().modify("1").set_name(
    original_cost_center.name
).save()
